presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1104,34 +1107,32 @@ else if (criteria instanceof JoinOn) {
 
             for (Expression conjunct : ExpressionUtils.extractConjuncts((Expression) optimizedExpression)) {
                 conjunct = ExpressionUtils.normalize(conjunct);
```

Comment:
`normalize` on `NOT a.x = b.x` will result in `a.x != b.x` which will still go into `postInnerJoin`, unless I'm mistaken (see line 288 in `RelationPlanner`. It looks like `ComparisonExpression.Type.EQUAL` is the only type of predicate which has efficient handling